src/main/java/gregtech/api/worldgen/config/WorldGenRegistry.java

```diff
@@ -93,13 +93,15 @@ public void reinitializeRegisteredVeins() throws IOException {
         oreVeinCache.clear();
         Path configPath = Loader.instance().getConfigDir().toPath().resolve(GTValues.MODID);
         Path worldgenRootPath = configPath.resolve("worldgen");
+        Path jarFileExtractLockOld = configPath.resolve(".worldgen_extracted");
         Path jarFileExtractLock = configPath.resolve("worldgen_extracted");
         if(!Files.exists(worldgenRootPath)) {
             Files.createDirectories(worldgenRootPath);
         }
         //attempt extraction if file extraction lock is absent or worldgen root directory is empty
-        if(!Files.exists(jarFileExtractLock) || !Files.list(worldgenRootPath).findFirst().isPresent()) {
-            if(!Files.exists(jarFileExtractLock)) {
+        // todo: "Old" is fugly.  Deprecate and/or replace with pattern-match on name
+        if((!Files.exists(jarFileExtractLock) && !Files.exists(jarFileExtractLockOld)) || !Files.list(worldgenRootPath).findFirst().isPresent()) {
+            if(!Files.exists(jarFileExtractLock) && !Files.exists(jarFileExtractLockOld)) {
```

Comment:
Wrong. Second if condition here shouldn't be the case.

Reply:
Archengius: that's actually your own secondary if condition, sir.  If we remove it, then we need to also remove your OR logic that checks for an empty folder and creates the file.

Otherwise, if Files.list(worldgenRootPath).findFirst().isPresent() qualifies, we'll attempt to create the file.  If you're ok with that, I'll remove the condition.

Reply:
do you understand that with !Files.exists(jarFileExtractLockOld)) { it will not generate new lock file when there is deprecated one, resulting in worldgen extraction when support for deprecated file is removed,